rasterio/_base.pyx

```diff
@@ -389,9 +389,10 @@ cdef class DatasetReader(object):
             row += self.height
         return c+a*col, f+e*row
 
-    def index(self, x, y, op=math.floor):
-        """Returns the (row, col) index of the pixel containing (x, y)."""
-        return get_index(x, y, self.affine)
+    def index(self, x, y, **kwargs):
```

Comment:
@jdmcbr in this case let's use `op` and `precision` keyword arguments instead of `**kwargs` if you don't mind. If you're quick, I can get this in the 0.28 release.